--- bluez-5.13/btio/btio.c	2013-12-28 01:16:57.000000000 +0800
+++ btio.c	2014-08-12 15:41:33.522787063 +0800
@@ -977,11 +977,18 @@
 
 	if (src.l2_bdaddr_type != BDADDR_BREDR) {
 		if (getsockopt(sock, SOL_BLUETOOTH, BT_RCVMTU,
-							&l2o.imtu, &len) < 0) {
-			ERROR_FAILED(err, "getsockopt(BT_RCVMTU)", errno);
-			return FALSE;
+							&l2o.imtu, &len) == 0) {
+			goto parse_opts;
+		}
+		else {
+			/* Non-LE CoC enabled kernels will return one of these
+			 * in which case we need to fall back to L2CAP_OPTIONS.
+			 */
+			if (errno != EPROTONOSUPPORT && errno != ENOPROTOOPT) {
+				ERROR_FAILED(err, "getsockopt(BT_RCVMTU)", errno);
+				return FALSE;
+			}
 		}
-		goto parse_opts;
 	}
 
 	if (getsockopt(sock, SOL_L2CAP, L2CAP_OPTIONS, &l2o, &len) < 0) {
